tools/bisection/perf-matrix.py

```
@@ -0,0 +1,369 @@
+
+    print("| Metric | libpinyin KC | libpinyin Tkrzw | oxpinyin KC | oxpinyin Tkrzw |")
+    print("|---|---:|---:|---:|---:|")
+    for name, fn in speed_metrics:
+        vals = [fn(speed_cells[l]) for l in CELL_MATRIX_ORDER]
+        print(f"| {name} | {vals[0]:.3f} | {vals[1]:.3f} | {vals[2]:.3f} | {vals[3]:.3f} |")
+    for name, fn in ram_metrics:
+        vals = [fn(ram_cells[l]) for l in CELL_MATRIX_ORDER]
+        print(f"| {name} | {vals[0]:,.0f} | {vals[1]:,.0f} | {vals[2]:,.0f} | {vals[3]:,.0f} |")
+    if size_cells:
+        for name, key in [(".so (KiB)", "so"), ("Data (KiB)", "data")]:
+            vals = [size_cells.get(l, {}).get(key, 0) / 1024 for l in CELL_MATRIX_ORDER]
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)